src/biogeophys/CanopyFluxesMod.F90

```diff
@@ -1049,6 +1066,17 @@ subroutine CanopyFluxes(bounds,  num_exposedvegp, filter_exposedvegp,
             dt_veg(p) = (sabv(p) + air(p) + bir(p)*t_veg(p)**4 + &
                  cir(p)*lw_grnd - efsh - efe(p)) / &
                  (- 4._r8*bir(p)*t_veg(p)**3 +dc1*wtga +dc2*wtgaq*qsatldT(p))
+            temp = dt_veg(p)
+            dt_veg(p) = ((1._r8-frac_rad_abs_by_stem)*(sabv(p) + air(p) &
+                  + bir(p)*t_veg(p)**4 + cir(p)*lw_grnd) &
+                  - efsh - efe(p) - lw_leaf + lw_stem &
+                  - (cp_leaf/dtime)*(t_veg(p) - tl_ini)) &
+                  / ((1._r8-frac_rad_abs_by_stem)*(- 4._r8*bir(p)*t_veg(p)**3 &
+                  + 4._r8*sa_internal*emv(p)*sb*t_veg(p)**3 &
+                  +dc1*wtga+dc2*wtgaq*qsatldT(p))+ cp_leaf/dtime)
+            if ( abs(temp - dt_veg(p) ) > 1.e-12 )then
+               call endrun( "Difference greater than roundoff" )
+            end if
```

Comment:
@ekluzek - Thanks for all of your work to demonstrate only roundoff-level diffs. I know from my own experience how painful that can sometimes be.

One question: are the terms you're checking here all order 1 (in magnitude)? If not, or if you're not sure, I generally use relative diff checks for things like this:

```Fortran
if (abs(temp - dt_veg(p)) > 1.e-13_r8 * temp) then
```

Reply:
That's a good point. Actually I'm sure they aren't order 1. dt_veg and err will be order 1, but could also be very small. ulrad and dlrad should be order 100. So doing a relative diff would be a better comparison. If I do that I'm likely to be able to show a relative difference that's a bit smaller as well.

Reply:
No need to redo things, though, if you have already done it this way and you have a general sense of the magnitude. My main worry with abs diffs is if the terms are very small to begin with - so an abs diff of 1e-12 could be a relative diff of, say, 1e-6. If you're pretty confident that isn't generally the case here, then I think this is good enough.

Reply:
I went ahead and did a little testing with this to show it's fine. Relative diff's potentially a bit more troublesome if temp is identically zero, or negative. But with a relative diff I was able to show a tolerance of e-14 was fine for two tests.

Reply:
> Relative diff's potentially a bit more troublesome if temp is identically zero, or negative.

Just a thought for the future: If `temp` is the old (presumed correct) version, then I usually have success with something like `if (abs(temp - dt_veg(p)) > 1.e-13_r8 * temp)` – or change the last `temp` to `abs(temp)` if negative values are acceptable. This avoids divide by 0. If temp (the old version) was exactly 0 at a point, then I typically expect the new version to be exactly zero, too, and I generally want to ensure that's true (which is done by the above) – since exactly 0 vs. slightly different from 0 can sometimes lead to large downstream differences.